pkg/signup/service/signup_service.go

```diff
@@ -438,7 +438,8 @@ func (s *ServiceImpl) DoGetSignup(ctx *gin.Context, provider ResourceProvider, u
 	// Check UserSignup status to determine whether user signup is complete
 	approvedCondition, approvedFound := condition.FindConditionByType(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupApproved)
 	completeCondition, completeFound := condition.FindConditionByType(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
-	if !approvedFound || !completeFound || approvedCondition.Status != apiv1.ConditionTrue {
+	if !approvedFound || !completeFound ||
+		(approvedCondition.Status != apiv1.ConditionTrue && approvedCondition.Reason != toolchainv1alpha1.UserSignupUserDeactivatedReason) {
```

Comment:
this would mean that if the approved is found, but the status is not true (is pending approval), then the if statement would be evaluated as "false", which would mean (in this context) as approved, which is wrong, right?
We want to mark the UserSignup as pending approval when:
1. approved condition is either not found or the status is false
2. or the complete condition is not found (it's almost the same as the previous prerequisite, but let's keep it there to be 100% sure)UserSignupUserDeactivatedReason
3. and at the same time the reason in approved condition is not `UserSignupUserDeactivatedReason`

It shouldn't happen that the approved condition would have a reason  `UserSignupUserDeactivatedReason` and the complete condition wouldn't be found, so no need to deal with that situation.

Please try to play with the `condition.` package - you may found some useful functions there that could simplify it.

Reply:
I'm idiot :man_facepalming: 
It's Monday and I cannot read the if statement correctly. Your version should work fine.
Could you please add/update some unit tests?

Reply:
Have updated the tests.

Reply:
Also changed the if statement per Alexey's comment .